multi_partner_learning.py

```diff
@@ -62,12 +62,12 @@ def __init__(self,
         self.aggregation_weights = []
         self.models_weights_list = [None] * self.partners_count
         self.scores_last_learning_round = [None] * self.partners_count
-        self.score_matrix_per_partner = np.zeros(shape=(self.epoch_count, self.minibatch_count, self.partners_count))
-        self.score_matrix_collective_models = np.zeros(shape=(self.epoch_count, self.minibatch_count + 1))
+        self.score_matrix_per_partner = np.nan * np.zeros(shape=(self.epoch_count, self.minibatch_count, self.partners_count))
```

Comment:
Before the score matrix was initialized with zero, now it is initialized with NaN.
Because of early stopping, every entry of the matrix will not be written. It makes for sense for these entries to be NaN that zero.

Reply:
Ok